### PR TITLE
[AArch64] Add intrinsics for multi-vector to ZA array vector accumulators

### PR DESCRIPTION
**Pull request by momchil-velikov**
According to the specification in
https://github.com/ARM-software/acle/pull/309 this adds the intrinsics

    void_svadd_za16_vg1x2_f16(uint32_t slice, svfloat16x2_t zn) __arm_streaming __arm_inout("za");
    void_svadd_za16_vg1x4_f16(uint32_t slice, svfloat16x4_t zn) __arm_streaming __arm_inout("za");
    void_svsub_za16_vg1x2_f16(uint32_t slice, svfloat16x2_t zn) __arm_streaming __arm_inout("za");
    void_svsub_za16_vg1x4_f16(uint32_t slice, svfloat16x4_t zn) __arm_streaming __arm_inout("za");

as well as the corresponding `bf16` variants.

### CODE REVIEW

File: clang/include/clang/Basic/arm_sme.td [hunk 1]
```diff
@@ -298,6 +298,17 @@ multiclass ZAAddSub<string n_suffix> {
     def NAME # _ZA64_VG1X2_F64 : Inst<"sv" # n_suffix # "_za64[_{d}]_vg1x2", "vm2", "d", MergeNone, "aarch64_sme_" # n_suffix # "_za64_vg1x2", [IsStreaming, IsInOutZA], []>;
     def NAME # _ZA64_VG1X4_F64 : Inst<"sv" # n_suffix # "_za64[_{d}]_vg1x4", "vm4", "d", MergeNone, "aarch64_sme_" # n_suffix # "_za64_vg1x4", [IsStreaming, IsInOutZA], []>;
   }
+
+  let TargetGuard = "sme2p1,sme-f16f16" in {
+    def NAME # _ZA16_VG1X2_F16 : Inst<"sv" # n_suffix # "_za16[_{d}]_vg1x2", "vm2", "h", MergeNone, "aarch64_sme_" # n_suffix # "_za16_vg1x2", [IsStreaming, IsInOutZA], []>;
+    def NAME # _ZA16_VG1X4_F16 : Inst<"sv" # n_suffix # "_za16[_{d}]_vg1x4", "vm4", "h", MergeNone, "aarch64_sme_" # n_suffix # "_za16_vg1x4", [IsStreaming, IsInOutZA], []>;
+  }
+
+  let TargetGuard = "sme2p1,b16b16" in {
```
[the remark this answers is not in the document]
Comment (momchil-velikov) [reply clang/include/clang/Basic/arm_sme.td:307]: Yes, when we fix these to the correct features, I'll rebase this patch.

File: clang/test/CodeGen/aarch64-sme2-intrinsics/acle_sme2_add_sub_za16.c [hunk 2]
```diff
@@ -0,0 +1,191 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 4
+// RUN: %clang_cc1                               -fclang-abi-compat=latest -triple aarch64-none-linux-gnu -target-feature +sme2p1  -target-feature +sme-f16f16 -target-feature +b16b16 -O2 -S -Werror -Wall -emit-llvm -o - %s | FileCheck %s
+// RUN: %clang_cc1                        -x c++ -fclang-abi-compat=latest -triple aarch64-none-linux-gnu -target-feature +sme2p1  -target-feature +sme-f16f16 -target-feature +b16b16 -O2 -S -Werror -Wall -emit-llvm -o - %s | FileCheck %s -check-prefix CHECK-CXX
+// RUN: %clang_cc1 -DSVE_OVERLOADED_FORMS        -fclang-abi-compat=latest -triple aarch64-none-linux-gnu -target-feature +sme2p1  -target-feature +sme-f16f16 -target-feature +b16b16 -O2 -S -Werror -Wall -emit-llvm -o - %s | FileCheck %s
+// RUN: %clang_cc1 -DSVE_OVERLOADED_FORMS -x c++ -fclang-abi-compat=latest -triple aarch64-none-linux-gnu -target-feature +sme2p1  -target-feature +sme-f16f16 -target-feature +b16b16 -O2 -S -Werror -Wall -emit-llvm -o - %s | FileCheck %s -check-prefix CHECK-CXX
+
+// REQUIRES: aarch64-registered-target
```
[the remark this answers is not in the document]
Comment (momchil-velikov) [reply clang/test/CodeGen/aarch64-sme2-intrinsics/acle_sme2_add_sub_za16.c:7]: I don't think so.

**Review by CarolineConcatto — APPROVED** [review]

**Comment by nico** [comment]
Thanks for the quick revert!

Is the failure due to a conflict with another commit that landed?

**Comment by momchil-velikov** [comment]
> Thanks for the quick revert!
> 
> Is the failure due to a conflict with another commit that landed?

Perhaps, e.g. https://github.com/llvm/llvm-project/pull/91140

**Pull request merged**